sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,280 @@
+        db.dataSource,
+        {
+          student,
+          msfaaNumber: currentMSFAA,
+        },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          offeringIntensity: OfferingIntensity.fullTime,
+          createSecondDisbursement: true,
+          disbursementScheduleStatusA: DisbursementScheduleStatus.Sent,
+          disbursementScheduleStatusB: DisbursementScheduleStatus.Pending,
+        },
+      );
+      const applicationB = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          msfaaNumber: currentMSFAA,
+        },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          offeringIntensity: OfferingIntensity.fullTime,
+          createSecondDisbursement: true,
+          disbursementScheduleStatusA: DisbursementScheduleStatus.Pending,
+          disbursementScheduleStatusB: DisbursementScheduleStatus.Sent,
+        },
+      );
+      // Queued job.
+      const job = createMock<Job<void>>();
+      mockDownloadFiles(sftpClientMock, [
+        MSFAA_FULL_TIME_RECEIVE_FILE_WITH_REACTIVATION_RECORD,
+      ]);
+      // Act
+      //Now reactivate the cancelled MSFAA
+      const processResult = await processor.processMSFAA(job);
+      // Assert
+      expect(processResult).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${MSFAA_FULL_TIME_RECEIVE_FILE_WITH_REACTIVATION_RECORD}.`,
+            "File contains:",
+            "Confirmed MSFAA records (type R): 1.",
+            "Cancelled MSFAA records (type C): 0.",
+            "Record from line 1, updated as confirmed.",
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      // Assert that the file was deleted from SFTP.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+      // Find the updated MSFAA records previously created.
+      const msfaaIDs = createdMSFAARecords.map((msfaa) => msfaa.id);
+      const [createdReactivatedMsfaaRecord] = createdMSFAARecords;
+      const msfaaUpdatedRecords = await db.msfaaNumber.find({
+        select: {
+          msfaaNumber: true,
+          dateSigned: true,
+          serviceProviderReceivedDate: true,
+          cancelledDate: true,
+          newIssuingProvince: true,
+        },
+        where: {
+          id: In(msfaaIDs),
+        },
+        order: {
+          msfaaNumber: "ASC",
+        },
+      });
+      expect(msfaaUpdatedRecords).toHaveLength(msfaaInputData.length);
+      const [reactivatedMSFAA] = msfaaUpdatedRecords;
+      // Validate reactivated confirmed record.
+      expect(reactivatedMSFAA.dateSigned).toBe("2023-05-02");
+      expect(reactivatedMSFAA.serviceProviderReceivedDate).toBe("2023-05-03");
+      expect(reactivatedMSFAA.cancelledDate).toBe(null);
+      // Validate msfaa updated to the reactivated msfaa for all pending disbursements.
+      const [appAFirstDisbursementSchedule, appASecondDisbursementSchedule] =
```

Comment:
the suggestion for this type of data preparation of creating 2 applications with 2 disbursements each was given by @andrewsignori-aot

Reply:
@dheepak-aot the process must ensure that other applications would also be updated.
Since we were creating just one E2E I suggested having applications A and B.
We should have even more applications to ensure that applications from the same student with different intensities would not be impacted, for instance.